File: apps/cms-react/src/services/userService.js
import api from './api'

export const userService = {
  // Get all users
  getAll: async (skip = 0, limit = 100) => {
    const response = await api.get('/api/users', {
      params: { skip, limit },
    })
    return response.data
  },

  // Get user by ID
  getById: async (id) => {
    const response = await api.get(`/api/users/${id}`)
    return response.data
  },

  // Create new user
  create: async (userData) => {
    const response = await api.post('/api/users', userData)
    return response.data
  },

  // Update user
  update: async (id, userData) => {
    const response = await api.put(`/api/users/${id}`, userData)
    return response.data
  },

  // Delete user
  delete: async (id) => {
    await api.delete(`/api/users/${id}`)
  },
}

File: apps/backend/src/inorta_backend/services/user_service.py
from sqlalchemy.orm import Session
from typing import List, Optional

from inorta_backend.models.user import User
from inorta_backend.schemas.user import UserCreate, UserUpdate


class UserService:
    """Service for User CRUD operations"""

    @staticmethod
    def get_users(db: Session, skip: int = 0, limit: int = 100) -> List[User]:
        """Get all users with pagination"""
        return db.query(User).offset(skip).limit(limit).all()

    @staticmethod
    def get_user_by_id(db: Session, user_id: int) -> Optional[User]:
        """Get user by ID"""
        return db.query(User).filter(User.id == user_id).first()

    @staticmethod
    def get_user_by_email(db: Session, email: str) -> Optional[User]:
        """Get user by email"""
        return db.query(User).filter(User.email == email).first()

    @staticmethod
    def create_user(db: Session, user: UserCreate) -> User:
        """Create a new user"""
        db_user = User(
            email=user.email,
            name=user.name
        )
        db.add(db_user)
        db.commit()
        db.refresh(db_user)
        return db_user

    @staticmethod
    def update_user(db: Session, user_id: int, user: UserUpdate) -> Optional[User]:
        """Update an existing user"""
        db_user = db.query(User).filter(User.id == user_id).first()
        if not db_user:
            return None
        
        update_data = user.model_dump(exclude_unset=True)
        for field, value in update_data.items():
            setattr(db_user, field, value)
        
        db.commit()
        db.refresh(db_user)
        return db_user

    @staticmethod
    def delete_user(db: Session, user_id: int) -> bool:
        """Delete a user"""
        db_user = db.query(User).filter(User.id == user_id).first()
        if not db_user:
            return False
        
        db.delete(db_user)
        db.commit()
        return True
